src/v/raft/types.h

```diff
@@ -326,6 +326,8 @@ struct batch_identity {
     producer_identity pid;
     int32_t seq{0};
     int32_t record_count{0};
+
+    bool has_idempotent() { return pid.id >= 0; }
```

Comment:
at the raft layer is it true that this is describing an idempotent property of the batch? it seems like the idempotent-ness is a property that exists above raft in kafka?

Reply:
right,moved it to `model/record.h`